src/Simplex/Messaging/Agent/Store.hs

```diff
@@ -1,31 +1,52 @@
 {-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DuplicateRecordFields #-}
 {-# LANGUAGE GADTs #-}
 {-# LANGUAGE KindSignatures #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE StandaloneDeriving #-}
 {-# OPTIONS_GHC -fno-warn-unticked-promoted-constructors #-}
 
-module Simplex.Messaging.Agent.Store
-  ( ReceiveQueue (..),
-    SendQueue (..),
-    Connection (..),
-    SConnType (..),
-    SomeConn (..),
-    MessageDelivery (..),
-    DeliveryStatus (..),
-    MonadAgentStore (..),
-  )
-where
+module Simplex.Messaging.Agent.Store where
 
+import Control.Exception (Exception)
+import Data.Int (Int64)
 import Data.Kind (Type)
-import Data.Time.Clock (UTCTime)
+import Data.Time (UTCTime)
 import Data.Type.Equality
-import Simplex.Messaging.Agent.Store.Types (ConnType (..))
 import Simplex.Messaging.Agent.Transmission
 import qualified Simplex.Messaging.Protocol as SMP
-import Simplex.Messaging.Types (RecipientPrivateKey, SenderPrivateKey, SenderPublicKey)
+import Simplex.Messaging.Types
+  ( MsgBody,
+    MsgId,
+    RecipientPrivateKey,
+    SenderPrivateKey,
+    SenderPublicKey,
+  )
+
+-- * Store management
+
+-- | Store class type. Defines store access methods for implementations.
+class Monad m => MonadAgentStore s m where
+  -- Queue and Connection management
+  createRcvConn :: s -> ReceiveQueue -> m ()
+  createSndConn :: s -> SendQueue -> m ()
+  getConn :: s -> ConnAlias -> m SomeConn
+  getRcvQueue :: s -> SMPServer -> SMP.RecipientId -> m ReceiveQueue
+  deleteConn :: s -> ConnAlias -> m ()
+  upgradeRcvConnToDuplex :: s -> ConnAlias -> SendQueue -> m ()
+  upgradeSndConnToDuplex :: s -> ConnAlias -> ReceiveQueue -> m ()
+  setRcvQueueStatus :: s -> ReceiveQueue -> QueueStatus -> m ()
+  setSndQueueStatus :: s -> SendQueue -> QueueStatus -> m ()
 
+  -- Msg management
+  createRcvMsg :: s -> ConnAlias -> MsgBody -> ExternalSndId -> ExternalSndTs -> BrokerId -> BrokerTs -> m ()
```

Comment:
this method should return something - most likely InternalId - because the agent should sent to the client message with internally assigned ID (currently it just slaps 0). It can be in the next PR.

Client then would use this ID to refer to this message in other commands

Reply:
good catch, for some reason it slipped my mind. will change in separate PR